doc/examples/decomposition/four-block-tree.cpp

```diff
@@ -0,0 +1,49 @@
+#include <ogdf/basic/graph_generators/randomized.h>
+#include <ogdf/basic/simple_graph_alg.h>
+#include <ogdf/decomposition/FourBlockTree.h>
+#include <ogdf/fileformats/GraphIO.h>
+#include <ogdf/planarlayout/PlanarStraightLayout.h>
+
+#include <string>
+
+using namespace ogdf;
+
+int main(void) {
+	Graph g;
+	constexpr int n = 16;
+	randomPlanarConnectedGraph(g, n, 3 * n - 6);
+	const adjEntry externalFace = g.firstNode()->firstAdj()->cyclicSucc();
+
+	const FourBlockTree fbt = FourBlockTree::construct(g, externalFace);
+
+	PlanarStraightLayout layout;
+	{
+		GraphAttributes ga(g,
+				ogdf::GraphAttributes::nodeLabel | ogdf::GraphAttributes::nodeGraphics
+						| ogdf::GraphAttributes::nodeStyle | ogdf::GraphAttributes::edgeGraphics
+						| ogdf::GraphAttributes::edgeStyle);
+		ga.directed() = false;
+		layout.callFixEmbed(ga, externalFace);
+		for (const node v : g.nodes) {
+			ga.label(v) = std::to_string(v->index());
+			ga.fillColor(v) = Color::Name::White;
+		}
+		GraphIO::write(ga, "output-g.svg", GraphIO::drawSVG);
+	}
+	int i = 0;
+	fbt.preorder([&](const FourBlockTree& treeNode) -> void {
+		GraphAttributes ga(treeNode.g,
+				ogdf::GraphAttributes::nodeLabel | ogdf::GraphAttributes::nodeGraphics
+						| ogdf::GraphAttributes::nodeStyle | ogdf::GraphAttributes::edgeGraphics
+						| ogdf::GraphAttributes::edgeStyle);
+		ga.directed() = false;
+		layout.callFixEmbed(ga, treeNode.externalFace);
```

Comment:
On my Windows machine, I get an exception here because `&ga.constGraph() != treeNode.externalFace->graphOf()`, which is weird, as that value should always be correctly initialized. The problem is probably somewhat contrived because it comes from an entirely different place. In your construction process, you return an instance of FourBlockTree in a few places, e.g. the `static FourBlockTree FourBlockTree::construct(...)`. Here, newer C++ standards (namely 17) mandate that the returned value is never actually copied/moved out of the function, but always constructed right in the place the caller wants it (because otherwise the by-value semantics that are a common bug with argument being inadvertedly copied would also apply for return values). This is called [guaranteed copy elision](https://philipp-jung.de/post/cpp_copy_elision/). Well guaranteed is seemingly not actually true, because setting `FourBlockTree(FourBlockTree&&) = delete` will fail the Windows Debug build in the exact places where the value is returned, while I'm pretty sure that gcc and clang correctly work around this even on debug builds. And the thing is that the `Graph` class is actually in deed not truly moveable, as it would need to update all `graphOf()` pointers and more when it would be moved. Unfortunately I could not yet officially mark the move constructor as deleted due to some very weird legacy code within the OGDF (that surely cannot work for debug builds and probably also fails for release builds, but that's another story).

So, to fix this we need to stop moving `Graph`s. This can either be done by refactoring your code to never return a `FourBlockTree` but construct it in-place (allowing it to be properly marked as non-moveable), or (maybe less work) to store the `Graph`s contained within the `FourBlockTree` either in a pointer that is deleted in the destructor or, even simpler, an `unique_ptr`.